server/src/main/java/io/crate/execution/engine/aggregation/impl/CountAggregation.java

```diff
@@ -276,4 +283,31 @@ public MutableLong removeFromAggregatedState(RamAccounting ramAccounting,
         }
         return null;
     }
+
+    @Nullable
+    @Override
+    public DocValueAggregator<?> getDocValueAggregator(List<Symbol> aggregationReferences,
+                                                       java.util.function.Function<List<String>, List<MappedFieldType>> getMappedFieldTypes,
+                                                       DocTableInfo table,
+                                                       List<Literal<?>> optionalParams) {
+        if (aggregationReferences.size() == 1 && aggregationReferences.get(0).valueType().id() == ObjectType.ID && table != null) {
+            var allNotNullSubColumns = new ArrayList<>(table.notNullColumns());
+            for (var notNullCol : allNotNullSubColumns) {
+                if (notNullCol.isChildOf(((Reference)aggregationReferences.get(0)).column())) {
```

Comment:
Maybe assign `((Reference)aggregationReferences.get(0)).column()` to a local variable outside of the loop, so that it doesn't have to do the element lookup on each iteration.